### PR TITLE
Uniqueindexes

### PR DESCRIPTION
**Pull request by twillouer**
A simple implementation for uniques indexes.

Only on a simple level, not complex (like "product.quantity").

Any idea to do better ?

### CODE REVIEW

File: src/main/java/com/foursquare/fongo/impl/IndexUtil.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+/**
+ * Class used to help with indexes.
+ */
+public final class IndexUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(IndexUtil.class);
+
+  private IndexUtil() {
+  }
+
+  private static List<List<Object>> extractFields(Iterable<DBObject> objects, List<String> fields) {
```
Comment (hoffrocket) [inline src/main/java/com/foursquare/fongo/impl/IndexUtil.java]: what you want here is ExpressionParser.getEmbeddedValues

**Comment by hoffrocket** [comment]
Thanks for this effort, but I'd like to implement indexes as real indexes, not just for uniqueness checks.  I think this type of change would adequately address the uniqueness guarantees, but it would come at a big performance cost. When I first implemented fongo I did it without the _id->dbo map and just did a linear scan on every find and the my unit tests took a performance hit compared to real mongo, so I implemented the hash map.

**Comment by hoffrocket** [comment]
Thanks for the work you've been doing. As a general note, I would rather keep unit tests written in java unless they they are specifically testing scala interoperability.  The reason is that for me personally there's a bit of a context switching overhead in going back and forth and I'm less familiar with the testing libraries. The scala unit tests that I added were an attempt to duplicate some strange bugs.

**Comment by twillouer** [comment]
As you want :)

_but_ I think ScalaTest are easyer to implement and understand..

I make an another PR for the command deleteIndexes

**Comment by twillouer** [comment]
ok, I take index as real index and send another PR

**Pull request closed without merging**